pkg/operator/operator.go

```diff
@@ -728,7 +746,28 @@ func (o *Operator) sync(ctx context.Context, key string) error {
 		klog.Warningf("Fail to load ConsoleConfig, AlertManager's externalURL may be outdated")
 	}
 
-	factory := manifests.NewFactory(o.namespace, o.namespaceUserWorkload, config, o.loadInfrastructureConfig(ctx), proxyConfig, o.assets, apiServerConfig, consoleConfig)
+	// Enable kube-state-metrics' custom-resource-state-based metrics if VPA CRD is installed within the cluster.
+	enableKSMCRSMetrics := false
+	_, err = o.client.ApiExtensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Get(ctx, client.VerticalPodAutoscalerCRDMetadataName, metav1.GetOptions{})
```

Comment:
We might want to return an error if we get an error other than not found from the API, otherwise the operator might update the KSM deployment and revert the good config.
Something we did in other cases is to store the last know value in the operator struct and default to it in case of transient errors from the API.

(nit) I would encapsulate the logic in a new function in pkg/client (e.g. `func (c *Client) VerticalPodAutoscalerCRDPresent() (bool, error)`).